### PR TITLE
BUG: parse_dates=False while passing date_parser tries to use date parser

### LINKED ISSUE #44366: BUG: parse_dates=False while passing date_parser tries to use date parser
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
with open('tmp.csv', 'w+') as handle:
    handle.write('a;b;c\n0;0,00;0')

def parser(p):
    raise ValueError("Tried to use parser")

pd.read_csv('tmp.csv', date_parser=parser, parse_dates=False)
```

### Issue Description

Date parser gets run even when `parse_dates` is set to False. This happens because Line 510 in readers.py ([link](https://github.com/pandas-dev/pandas/blob/master/pandas/io/parsers/readers.py#L510) only checks whether `parse_dates`  is a bool rather than the value.
I've not been able to reproduce without the comma and semicolons in there. 

### Expected Behavior

Getting the same result as `pd.read_csv('tmp.csv')` which is a malformed DataFrame.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
python           : 3.8.10.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18363
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 4, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United Kingdom.1252
pandas           : 1.2.4
numpy            : 1.20.2
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.1.1
setuptools       : 52.0.0.post20210125
Cython           : None
pytest           : 6.2.4
hypothesis       : 6.14.1
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : None
pyxlsb           : 1.0.8
s3fs             : None
scipy            : None
sqlalchemy       : 1.4.22
tables           : None
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by loicdiridollou**
- [x] closes #44366 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

I have updated the code in the `pandas/io/parsers/readers.py` since if we pased `parse_dates=False`, it would still change it to True.

### CODE REVIEW

File: pandas/io/parsers/readers.py [hunk 1]
```diff
@@ -511,8 +511,14 @@ def _read(
 ):
     """Generic reader of line files."""
     if kwds.get("date_parser", None) is not None:
```
Comment (phofl) [inline pandas/io/parsers/readers.py:513]: This is hard to read. Please do it somehow like this:

```
if kwds.get("date_parser", None) is not None and kwds.get("pars_dates", None) is None:
    kwds.get["pars_dates"] = True
elif kwds.get("pars_dates", None) is None:
    kwds.get["pars_dates"] = False
```

**Review by phofl — CHANGES_REQUESTED** [review]
Please add tests. One whatsnew is enough

File: pandas/io/parsers/readers.py (hunk 1 above)
Comment (loicdiridollou) [reply pandas/io/parsers/readers.py]: Correct! I have made the changes so now it is more readable, added a test to see if the parser is not called when we set `parse_dates` to `False`.
Some tests are failing for an obscure reason as they are not related to the code change (and they are also failing on other builds)

File: pandas/tests/io/parser/test_parse_dates.py [hunk 2]
```diff
@@ -97,6 +97,43 @@ def __custom_date_parser(time):
...
+def test_read_csv_with_custom_date_parser_parse_dates_false(all_parsers):
+    # GH44366
+    def __custom_date_parser(time):
+        time = time.astype(np.float_)
+        time = time.astype(np.int_)  # convert float seconds to int type
+        return pd.to_timedelta(time, unit="s")
+
+    testdata = StringIO(
+        """time e n h
+        41047.00 -98573.7297 871458.0640 389.0089
```
Comment (phofl) [inline pandas/tests/io/parser/test_parse_dates.py:110]: Can we reduce number of columns and use small ints? This is really hard to read

Comment (loicdiridollou) [reply pandas/tests/io/parser/test_parse_dates.py]: You are right, this should be better now. I was just reusing the data from `test_read_csv_with_custom_date_parser` (which could also be simplified at a later stage)

**Review by jreback — APPROVED** [review]
lgtm. @phofl merge when good

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @loicdiridollou

**Pull request merged**